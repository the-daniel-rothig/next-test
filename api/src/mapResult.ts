export type Result = {
    kind: "Artist" | "Album" | "Song",
    name: string,
    url: string,
    artwork: string | null,
}

export const mapResult = (res: any): Result => {
    if (res.wrapperType === "artist") {
        return {
            kind: "Artist",
            name: res.artistName,
            url: res.artistViewUrl,
            artwork: null
        }
    }
    if (res.wrapperType === "collection") {
        return {
            kind: "Album",
            name: `${res.collectionName} by ${res.artistName}`,
            url: res.collectionViewUrl,
            artwork: res.artworkUrl60,
        }
    }
    if (res.wrapperType === "track") {
        return {
            kind: "Song",
            name: `${res.trackName} by ${res.artistName}`,
            url: res.trackViewUrl,
            artwork: res.artworkUrl60,
        }        
    }

    throw `Wrapper type not supported: ${res.wrapperType}`;
}