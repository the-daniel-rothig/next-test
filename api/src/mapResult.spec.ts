import { mapResult } from "./mapResult";

describe('mapResult', () => {
    it('maps artists correctly', async () => {
        const input = {
            wrapperType: 'artist',
            artistName: 'artistName',
            artistViewUrl: 'artistViewUrl'
        }

        expect(mapResult(input)).toMatchSnapshot();
    });

    it('maps albums correctly', async () => {
        const input = {
            wrapperType: 'collection',
            collectionName: 'collectionName',
            artistName: 'artistName',
            collectionViewUrl: 'collectionViewUrl',
            artworkUrl60: 'artworkUrl60'
        }

        expect(mapResult(input)).toMatchSnapshot();
    });
    it('maps tracks correctly', async () => {
        const input = {
            wrapperType: 'track',
            trackName: 'trackName',
            artistName: 'artistName',
            trackViewUrl: 'trackViewUrl',
            artworkUrl60: 'artworkUrl60'
        }

        expect(mapResult(input)).toMatchSnapshot();
    });
});
