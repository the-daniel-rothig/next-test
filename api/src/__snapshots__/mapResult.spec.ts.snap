// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapResult maps albums correctly 1`] = `
Object {
  "artwork": "artworkUrl60",
  "kind": "Album",
  "name": "collectionName by artistName",
  "url": "collectionViewUrl",
}
`;

exports[`mapResult maps artists correctly 1`] = `
Object {
  "artwork": null,
  "kind": "Artist",
  "name": "artistName",
  "url": "artistViewUrl",
}
`;

exports[`mapResult maps tracks correctly 1`] = `
Object {
  "artwork": "artworkUrl60",
  "kind": "Song",
  "name": "trackName by artistName",
  "url": "trackViewUrl",
}
`;
